filestore/test/test_maps_reader.py

```diff
@@ -0,0 +1,191 @@
+#   specific prior written permission.                                 #
+#                                                                      #
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS  #
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT    #
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS    #
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE       #
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,           #
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES   #
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR   #
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)   #
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,  #
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import six
+from metadatastore.api import (insert_run_start,
+                               insert_event,
+                               insert_event_descriptor,
+                               insert_beamline_config)
+import filestore.retrieve as fsr
+import filestore.commands as fsc
+
+from filestore.file_readers import HDFMapsSpectrumHandler as HDFM
+
+import h5py
+import os.path as op
+import numpy as np
+from nose.tools import assert_true, assert_raises, assert_false, assert_equal
+import datetime
+import uuid
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+fsr.register_handler('hdf_maps', HDFM)
+
+
+def save_syn_data(eid, data, base_path=None):
+    """
+    Save a array as hdf format to disk.
+    Defaults to saving files in :path:`~/.fs_cache/YYYY-MM-DD.h5`
+
+    Parameters
+    ----------
+    eid : unicode
+        id for file name
+    data : ndarray
+        The data to be saved
+    base_path : str, optional
+        The base-path to use for saving files.  If not given
+        default to `~/.fs_cache`.  Will add a sub-directory for
+        each day in this path.
+    """
+
+    if base_path is None:
+        base_path = op.join(op.expanduser('~'), '.fs_cache',
+                            str(datetime.date.today()))
+    fpath = op.join(base_path, str(eid) + '.h5')
+
+    with h5py.File(fpath, 'w') as f:
+        # create a group for maps to hold the data
+        mapsGrp = f.create_group('MAPS')
+        # now set a comment
+        mapsGrp.attrs['comments'] = 'MAPS group'
+
+        entryname = 'mca_arr'
+        comment = 'These are raw spectrum data.'
+        ds_data = mapsGrp.create_dataset(entryname, data=data)
+        ds_data.attrs['comments'] = comment
+    return fpath
+
+
+def get_data(ind_v, ind_h):
```

Comment:
It should be a simple find-and-replace on 3 functions (and 5 kwargs) so I
wouldn't worry about it too much.  sed is your friend ;)

On Sun Feb 22 2015 at 10:14:10 PM Li Li <notifications@github.com> wrote:

> In filestore/test/test_maps_reader.py
> <https://github.com/NSLS-II/filestore/pull/20#discussion_r25142325>:
>
> > +    fpath = op.join(base_path, str(eid) + '.h5')
> > +
> > +    with h5py.File(fpath, 'w') as f:
> > +        # create a group for maps to hold the data
> > +        mapsGrp = f.create_group('MAPS')
> > +        # now set a comment
> > +        mapsGrp.attrs['comments'] = 'MAPS group'
> > +
> > +        entryname = 'mca_arr'
> > +        comment = 'These are raw spectrum data.'
> > +        ds_data = mapsGrp.create_dataset(entryname, data=data)
> > +        ds_data.attrs['comments'] = comment
> > +    return fpath
> > +
> > +
> > +def get_data(ind_v, ind_h):
>
> I actually sent a PR to dataportal per @ericdill
> <https://github.com/ericdill> suggestion. It looks like a new version of
> filestore is coming. Shall I wait for the new version, @tacaswell
> <https://github.com/tacaswell> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NSLS-II/filestore/pull/20/files#r25142325>.
>